kedro/pipeline/modular_pipeline.py

```diff
@@ -2,174 +2,26 @@
 
 from __future__ import annotations
 
-import copy
-import difflib
 from typing import TYPE_CHECKING
 
-from kedro.pipeline.pipeline import Pipeline
+from kedro.pipeline.pipeline import (
+    Pipeline,
+    _get_dataset_names_mapping,
+    _get_param_names_mapping,
+    _is_all_parameters,
+    _is_single_parameter,
+    _validate_datasets_exist,
+    _validate_inputs_outputs,
+)
 
-from .transcoding import TRANSCODING_SEPARATOR, _strip_transcoding, _transcode_split
+from .transcoding import TRANSCODING_SEPARATOR, _transcode_split
 
 if TYPE_CHECKING:
-    from collections.abc import Iterable, Set
+    from collections.abc import Iterable
 
     from kedro.pipeline.node import Node
 
 
-class ModularPipelineError(Exception):
-    """Raised when a modular pipeline is not adapted and integrated
-    appropriately using the helper.
-    """
-
-    pass
```

Comment:
Removing this public class from the public module is technically a breaking change, right? If the plan described in https://github.com/kedro-org/kedro/discussions/4594 is to eventually rename this to `PipelineError`, is it worth moving the error like this in the `main` branch?

Reply:
See the todo's in the development notes for the PR 🙂  But also as I mentioned in stand-up, I'll probably create a new PR once the TSC approval has happened, so no need to review this thoroughly.